File: 10324.cpp
//UVa-10324             Zeros and Ones
//#include <bits/stdc++.h>
#include <algorithm>
#include <cstdio>
#include <string>
#include <cstring>
#include <map>
#include <vector>
#include <math.h>
#define LIM         1005000
#define loop(i,a,b) for(i=a;i<b;i++)
#define SET(a,b)    memset(a,b,sizeof a)
#define pb          push_back
using namespace std;

int main(){
  char cad[LIM];
  int s,f,q,i,casos=1,j,act;
  int acum[LIM];
  while(scanf("%s %d",cad,&q)==2 && cad!=NULL){
    acum[0]=cad[0]-'0';
    loop(i,1,(int)strlen(cad)) acum[i]=acum[i-1]+(cad[i]-'0');
    printf("Case: %d:\n",casos++);
    loop(i,0,q){
      scanf("%d %d",&s,&f);
      act=fabs(acum[f]-acum[s]);
      if(act==f-s || act==0) printf("Yes\n");
      else printf("No\n");
    }
  }
  return 0;
}

File: 10324.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.math.BigInteger;

public class Main {
  public static void main(String[] args) throws Exception {
    int q,s,f,act,casos=1,tem;
    int []acum;
    String cad;
    String[] tmp;
    BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
    while(br.ready()){
      cad=br.readLine().toString();
      if(cad==null) break;
      acum=new int[cad.length()];
      acum[0]=cad.charAt(0)-'0';
      for (int i = 1; i < cad.length(); i++) acum[i]=acum[i-1]+cad.charAt(i)-'0'; 
      q=Integer.parseInt(br.readLine().toString());
      System.out.println("Case "+(casos++)+":");
      for (int i = 0; i < q; i++) {
        tmp = br.readLine().toString().split(" ");
        s=Integer.parseInt(tmp[0]);
        f=Integer.parseInt(tmp[1]);
        if(s>f){ tem =f; f=s; s=tem; }
        if(s==0) act=acum[f];
        else act=acum[f]-acum[s-1];
          if(act==f-s+1 || act==0) System.out.println("Yes");
          else System.out.println("No");
      }
    }
  }
}

